src/suricata.c

```diff
@@ -2155,6 +2155,10 @@ static int InitRunAs(SCInstance *suri)
     }
     /* Get the suricata user ID to given user ID */
     if (suri->do_setuid == TRUE) {
+        if (suri->user_name == NULL) {
+            SCLogError("user name cannot be set to an empty value");
+            return TM_ECODE_FAILED;
+        }
```

Comment:
Should we check for empty value as well instead of just null?
```
./src/suricata -T --user ""
```

Reply:
I didn't alter the logic on that one because it was already sanitized with the message "unable to get the user ID, check if user exist!! "

Albeit, it is true that this message can be more informative.

Reply:
added a check for an empty user field